src/makeInfinityAdder.js

```diff
@@ -6,7 +6,21 @@
  */
 
 function makeInfinityAdder() {
-  // write code here
+  let temporarySum = 0;
+
+  return function add(input) {
+    if (typeof input === 'undefined') {
+      const temporarySumCopy = temporarySum;
+
+      temporarySum = 0;
+
+      return temporarySumCopy;
+    } else {
+      temporarySum += input;
+
+      return add;
+    }
```

Comment:
you don't need else here, anyways you return value from previous if
```suggestion
    }
   
    temporarySum += input;

    return add;
```